spec/controllers/books_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+describe BooksController, type: :controller do
+  let(:user) { create(:user) }
+
+  before { login(user) }
+
+  describe 'GET #index' do
+    context 'When fetching all the books' do
```

Comment:
I prefer having a global context that says something about having a current user and then add in that context the `before { login(user) }`